tasks/internal/get-advisory-severity/get-advisory-severity.yaml

```
@@ -119,6 +120,100 @@ spec:
               fi
           }
 
+          # Function to process a batch of CVEs with its own token
+          process_cve_batch() { # Expected arguments are [batch_id, batch_cves]
+              echo "Batch $1: Getting token..."
+              set +x
+              batch_token=$(curl --retry 3 --negotiate -u : "${OSIDB_URL}"/auth/token | jq -r '.access')
+              set -x
+
+              echo "Batch $1: Processing CVEs..."
+
+              while read -r cve; do
+                  if [[ -n "$cve" ]]; then
+                      cache_file="$CVE_CACHE_DIR/${cve}.json"
+
+                      # Note: If we bypass this check in a race condition, the output from OSIDB will be
+                      # the same anyways so it is safe - multiple batches writing identical data is harmless
+                      if [[ -f "$cache_file" ]]; then
+                          continue  # Already processed
+                      fi
+
+                      echo "Batch $1: Processing CVE ${cve}"
+                      curl_url="${OSIDB_URL}/osidb/api/v1/flaws?cve_id=${cve}&include_fields=${INCLUDE_FIELDS}"
+                      set +x
+
+                      # Try API call with current token, refresh and retry if it fails
+                      if ! curl --retry 3 \
+                          -H 'Content-Type: application/json' \
+                          -H "Authorization: Bearer ${batch_token}" \
+                          "$curl_url" | jq '.results[0]' > "$cache_file"; then
+                          echo "Batch $1: API call failed, refreshing token and retrying..."
+                          batch_token=$(curl --retry 3 --negotiate -u : "${OSIDB_URL}"/auth/token | jq -r '.access')
+                          curl --retry 3 -H 'Content-Type: application/json' -H "Authorization: Bearer ${batch_token}" \
+                              "$curl_url" | jq '.results[0]' > "$cache_file"
+                      fi
+                      set -x
```

jq '.results[0]' could write null to cache files if CVE not found
```suggestion
                      set -x
                      # Validate cache file was written with valid data
                      if [[ ! -s "$cache_file" ]]; then
                          echo "Batch $1: ERROR - Cache file for CVE ${cve} is empty" >&2
                          return 1
                      fi
                      # Check if the API returned null (CVE not found in OSIDB)
                      if [[ "$(cat "$cache_file")" == "null" ]]; then
                          echo "Batch $1: ERROR - CVE ${cve} not found in OSIDB" >&2
                          return 1
                      fi
```

---

Good catch and I verified we wouldn't fail here. Fixed!